include/os/freebsd/zfs/sys/zfs_ioctl_compat.h

```diff
@@ -37,6 +37,8 @@
 #include <sys/nvpair.h>
 #endif  /* _KERNEL */
 
+#define	ZFS_LEGACY_SUPPORT
```

Comment:
While I am fine with this, I want to ping a FreeBSD guy to review this to get him to approve. I am not sure which one to ping, so I am just going to ping a couple people whose names occurred to me first: @freqlabs @allanjude.

Reply:
I could pull this one if it seems better to wait.

Reply:
It should not be a problem. The FreeBSD guys are fairly quick to review FreeBSD patches.

Reply:
Let's at least add a comment before this `#define` explaining that these legacy interfaces are planned for removal in a future release.